imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -141,14 +229,21 @@ private ParseResults parseDetailRecords(File file, FileInfo fileInfo) {
 
             String line;
             while ((line = reader.readLine()) != null) {
-
                 try {
                     CallDetailRecordDto cdr = CsvHelper.csvLineToCdr(line);
 
-                    parseDetailRecord(cdr, records);
+                    if (!verifyOnly) {
+                        aggregateDetailRecord(cdr, records);
+                    }
 
                 } catch (IllegalArgumentException e) {
                     errors.add(String.format("Line %d: %s", lineNumber, e.getMessage()));
+                    errorCount++;
+                }
+                if (errorCount >= maxErrorCount) {
+                    errors.add(String.format("The maximum number of allowed errors of %d has been reached, " +
+                            "discarding all remaining errors.", maxErrorCount));
```

Comment:
Yes, so it doesn't "discard all remaining errors", it stops processing the file.  There may be many more errors, or there may be none.  We don't know since it stops processing.